File: JDK11/opentraceability/src/main/java/opentraceability/models/events/ErrorDeclaration.java
package opentraceability.models.events;

import opentraceability.interfaces.IEventKDE;
import opentraceability.utility.attributes.OpenTraceabilityArrayAttribute;
import opentraceability.utility.attributes.OpenTraceabilityAttribute;
import opentraceability.utility.attributes.OpenTraceabilityExtensionElementsAttribute;
import opentraceability.utility.attributes.OpenTraceabilityJsonAttribute;

import java.net.URI;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.ArrayList;

public class ErrorDeclaration {
    @OpenTraceabilityJsonAttribute(name="reason")
    @OpenTraceabilityAttribute(ns="", name="@type")
    public URI Reason = null;

    @OpenTraceabilityAttribute(ns="", name="declarationTime")
    public OffsetDateTime DeclarationTime = null;

    @OpenTraceabilityArrayAttribute(itemName="correctiveEventID")
    @OpenTraceabilityAttribute(ns="", name="correctiveEventIDs")
    public List<String> CorrectingEventIDs = null;

    @OpenTraceabilityExtensionElementsAttribute
    public List<IEventKDE> ExtensionKDEs = new ArrayList<IEventKDE>();
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/models/events/errordeclaration.kt
package models.events

import interfaces.IEventKDE
import utility.attributes.*
import java.util.*
import java.net.URI
import java.time.OffsetDateTime

class ErrorDeclaration {

    @OpenTraceabilityJsonAttribute("reason")
    @OpenTraceabilityAttribute("", "@type")
    var Reason: URI? = null

    @OpenTraceabilityAttribute("", "declarationTime")
    var DeclarationTime: OffsetDateTime? = null

    @OpenTraceabilityArrayAttribute("correctiveEventID")
    @OpenTraceabilityAttribute("", "correctiveEventIDs")
    var CorrectingEventIDs: ArrayList<String>? = null

    @OpenTraceabilityExtensionElementsAttribute
    var ExtensionKDEs: ArrayList<IEventKDE> = ArrayList<IEventKDE>()

}
